plugins/woocommerce/client/legacy/js/frontend/single-add-to-cart.js

```diff
@@ -0,0 +1,284 @@
+( function () {
+	function getFormData( form ) {
+		const formData = [];
+		const elements = form.querySelectorAll(
+			'input:not([name="product_id"]), select, button, textarea'
+		);
+
+		elements.forEach( ( element ) => {
+			const name = element.getAttribute( 'name' );
+
+			if ( ! name ) {
+				return;
+			}
+
+			let value = element.value;
+
+			if ( element.type === 'checkbox' ) {
+				value = element.checked ? value : '';
+			}
+
+			if ( Array.isArray( value ) ) {
+				value.forEach( ( val ) => {
+					formData.push( {
+						name: name,
+						value: val.replace( /\r?\n/g, '\r\n' ),
+					} );
+				} );
+			} else {
+				formData.push( {
+					name: name,
+					value: value.replace( /\r?\n/g, '\r\n' ),
+				} );
+			}
+		} );
```

Comment:
Super nitpick, but we could extract adding items to `dataForm` to a local function for clarity:

```suggestion
                const addToFormData = ( name, value ) => {
                		formData.push( {
				name,
				value: value.replace( /\r?\n/g, '\r\n' ),
			} );
                };

		elements.forEach( ( element ) => {
			const name = element.getAttribute( 'name' );

			if ( ! name ) {
				return;
			}

			let value = element.value;

			if ( element.type === 'checkbox' ) {
				value = element.checked ? value : '';
			}

			if ( Array.isArray( value ) ) {
				value.forEach( ( val ) => {
					addToFormData( name, val );
				} );
			} else {
				addToFormData( name, value );
			}
		} );
```